packages/sync-engine/src/database/postgres.ts

```diff
@@ -174,61 +175,202 @@ export class PostgresClient {
-       WHERE resource = $1`,
-      [resource]
+       WHERE resource = $1 AND "_account_id" = $2`,
+      [resource, accountId]
     )
   }
 
-  async markSyncError(resource: string, errorMessage: string): Promise<void> {
+  async markSyncError(resource: string, accountId: string, errorMessage: string): Promise<void> {
     await this.query(
       `UPDATE "${this.config.schema}"."_sync_status"
-       SET status = 'error', error_message = $2, updated_at = now()
-       WHERE resource = $1`,
-      [resource, errorMessage]
+       SET status = 'error', error_message = $3, updated_at = now()
+       WHERE resource = $1 AND "_account_id" = $2`,
+      [resource, accountId, errorMessage]
     )
   }
+
+  // Account management methods
+
+  async upsertAccount(accountData: {
+    id: string
+    raw_data: any // eslint-disable-line @typescript-eslint/no-explicit-any
+  }): Promise<void> {
+    const rawData = JSON.stringify(accountData.raw_data)
+
+    await this.query(
+      `INSERT INTO "${this.config.schema}"."accounts" ("id", "raw_data", "first_synced_at", "last_synced_at")
+       VALUES ($1, $2::jsonb, now(), now())
+       ON CONFLICT ("id")
+       DO UPDATE SET
+         "raw_data" = EXCLUDED."raw_data",
+         "last_synced_at" = now(),
+         "updated_at" = now()`,
+      [accountData.id, rawData]
+    )
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  async getAllAccounts(): Promise<any[]> {
+    const result = await this.query(
+      `SELECT raw_data FROM "${this.config.schema}"."accounts"
+       ORDER BY last_synced_at DESC`
+    )
+    return result.rows.map((row) => row.raw_data)
+  }
+
+  async getAccountRecordCounts(accountId: string): Promise<{ [tableName: string]: number }> {
+    const tables = [
+      'subscription_items',
+      'subscriptions',
+      'subscription_schedules',
+      'checkout_session_line_items',
+      'checkout_sessions',
+      'tax_ids',
+      'customers',
+      'charges',
+      'refunds',
+      'credit_notes',
+      'disputes',
+      'early_fraud_warnings',
+      'invoices',
+      'payment_intents',
+      'payment_methods',
+      'setup_intents',
+      'prices',
+      'plans',
+      'products',
+      'features',
+      'active_entitlements',
+      'reviews',
+      '_managed_webhooks',
+      '_sync_status',
+    ]
+
+    const counts: { [tableName: string]: number } = {}
+
+    for (const table of tables) {
+      const result = await this.query(
+        `SELECT COUNT(*) as count FROM "${this.config.schema}"."${table}"
+         WHERE "_account_id" = $1`,
+        [accountId]
+      )
+      counts[table] = parseInt(result.rows[0].count)
+    }
+
+    return counts
+  }
+
+  async deleteAccountWithCascade(
+    accountId: string,
+    useTransaction: boolean
+  ): Promise<{ [tableName: string]: number }> {
+    // Deletion order: dependencies first, then accounts last
+    const deletionOrder = [
+      'subscription_items',
```

Comment:
This list of tables is somewhat duplicated. We should probably have a single source of truth for tables.